Tests/ShamTests/Tests/StubRequestTests.swift

```diff
@@ -25,4 +25,32 @@ final class StubRequestTests: XCTestCase {
         let request = StubRequest(url: self.baseURLString)
         XCTAssertEqual(request.description, "ALL: \(self.baseURLString)")
     }
+    
+    // MARK: Query items tests
+    
+    func testAccuratelyValidatesWhenARequestHasAllQueryItems() {
+        // GIVEN: A request with many parameters.
+        let queryParametersURL = "\(self.baseURLString)?zebra=thing&aardvark=other_thing&giraffe=thing_3"
+        let request: StubRequest = .get(queryParametersURL)
+        
+        // GIVEN: A stub with fewer parameters.
+        let similarURL = "\(self.baseURLString)?aardvark=other_thing&zebra=thing"
+        let stub: StubRequest = .get(similarURL, validationRule: .allowMissingQueryParameters)
```

Comment:
Can `stub` and `request` be renamed to show they're similar? Right now they're named as if they're different objects, but they're both `StubRequest`, which makes reading the test a little difficult.